src/ResourceManager/Resources/Commands.Resources/ActiveDirectory/RemoveAzureADServicePrincipalCommand.cs

```diff
@@ -50,11 +55,20 @@ public override void ExecuteCmdlet()
                     ObjectId.ToString(),
                     () => servicePrincipal = ActiveDirectoryClient.RemoveServicePrincipal(ObjectId.ToString()));
 
+                if (RemoveApplication)
+                {
+                    Rest.Azure.OData.ODataQuery<Application> odataQueryFilter = new Rest.Azure.OData.ODataQuery<Application>();
+                    string appId = servicePrincipal.ApplicationId.ToString();
+                    odataQueryFilter = new Rest.Azure.OData.ODataQuery<Application>(a => a.AppId == appId);
+                    PSADApplication application = ActiveDirectoryClient.GetApplicationWithFilters(odataQueryFilter).First();
+                    ActiveDirectoryClient.RemoveApplication(application.ObjectId.ToString());
```

Comment:
This is an async operation, seems like we should also add an AsJob parameter here

Reply:
It looks like none of the Graph operation are marked as async (I've been checking by look for the begin logic).